rxandroidble/src/main/java/com/polidea/rxandroidble/internal/connection/RxBleConnectionConnectorImpl.java

```diff
@@ -51,71 +35,31 @@ public RxBleConnectionConnectorImpl(
             @Override
             public Observable<RxBleConnection> call() {
 
-                if (!rxBleAdapterWrapper.isBluetoothEnabled()) {
-                    return Observable.error(new BleDisconnectedException(bluetoothDevice.getAddress()));
-                }
-
                 final ConnectionComponent connectionComponent = connectionComponentBuilder.build();
                 RxBleRadioOperationConnect operationConnect = connectionComponent.connectOperationBuilder()
                         .setAutoConnect(autoConnect)
                         .build();
 
-                return enqueueConnectOperation(operationConnect)
-                        .flatMap(new Func1<BluetoothGatt, Observable<RxBleConnection>>() {
-                            @Override
-                            public Observable<RxBleConnection> call(final BluetoothGatt bluetoothGatt) {
-                                return Observable.merge(
-                                        justOnNext(connectionComponent.rxBleConnection()),
-                                        connectionComponent.gattCallback().<RxBleConnection>observeDisconnect()
-                                );
-                            }
-                        })
-                        .doOnUnsubscribe(disconnect(connectionComponent.disconnectOperation()));
+                return justOnNext(connectionComponent.rxBleConnection())
```

Comment:
This is equivalent (i think better readable)

```
rxBleRadio.queue(operationConnect)
                        .map(new Func1<BluetoothGatt, RxBleConnection>() {
                            @Override
                            public RxBleConnection call(BluetoothGatt bluetoothGatt) {
                                return connectionComponent.rxBleConnection();
                            }
                        })
```

Reply:
I have extracted variables so maybe it will be better to read.
I like that there is no need to create additional `Func1<BluetoothGatt, RxBleConnection>` — seems that `.delaySubscription()` is a perfect match